libr/util/buf.c

```diff
@@ -49,6 +49,9 @@ static RBufferSparse *sparse_append(RList *l, ut64 addr, const ut8 *data, int le
 	s->odata = NULL;
 	s->data = calloc (1, len);
 	if (!data) {
+		if (s->data) {
```

Comment:
I think the fix here should be different. We can check for `data` as first thing in the `sparse_append` function and return NULL if !data. After `s->data = calloc (1, len);` there should be a check for `!s->data`, in my opinion.